core/src/impl/Kokkos_Core.cpp

```diff
@@ -166,9 +215,7 @@ int get_ctest_gpu(const char* local_rank_str) {
   return std::stoi(id.c_str());
 }
 
-namespace {
-
-#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
+// extern function to extrat gpu # from args
 int get_gpu(const InitArguments& args) {
```

Comment:
Why did you remove this from an anonymous namespace?

Reply:
resolved with 424fc647

Reply:
I remember why I removed the anonymous namespace.  Added extern references from cuda and hip instance cpp's.

added it back in acf409e